wordpress_rock/rockcraft.yaml

```diff
@@ -0,0 +1,188 @@
+      a2enmod ssl
+      EOF
+  wordpress:
+    after:
+      - apache2
+    plugin: nil
+    build-environment:
+      - WP_PLUGINS: >-
+          404page
+          all-in-one-event-calendar
+          coschedule-by-todaymade
+          elementor
+          essential-addons-for-elementor-lite
+          favicon-by-realfavicongenerator
+          feedwordpress
+          genesis-columns-advanced
+          line-break-shortcode
+          no-category-base-wpml
+          post-grid
+          powerpress
+          redirection
+          relative-image-urls
+          rel-publisher
+          safe-svg
+          show-current-template
+          simple-301-redirects
+          simple-custom-css
+          so-widgets-bundle
+          svg-support
+          syntaxhighlighter
+          wordpress-importer
+          wp-font-awesome
+          wp-lightbox-2
+          wp-markdown
+          wp-mastodon-share
+          wp-polls
+          wp-statistics
+      - WP_VERSION: 5.9.3
+      - PHP_VERSION: 7.4
+      - IMAGE_RUN_USER_ID: 584792
+      - IMAGE_RUN_GROUP_ID: 584792
+    build-packages:
+      - bzr
+      - curl
+      - git
+      - unzip
+    override-build: |
+      curl -sSL --create-dirs https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar -o wp
+      chmod +x wp
+
+      mkdir -p wordpress_install_dir
+      (cd wordpress_install_dir
+      $CRAFT_PART_BUILD/wp core download --version=${WP_VERSION} --allow-root)
+
+      # Install wordpress plugins
+      (cd wordpress_install_dir/wp-content/plugins
+      for plugin in $WP_PLUGINS;
+        do
+          curl -sSL "https://downloads.wordpress.org/plugin/${plugin}.latest-stable.zip" -o "${plugin}.zip"
+          unzip "${plugin}.zip"
+          rm "${plugin}.zip"
+      done
+      curl -sSL "https://downloads.wordpress.org/plugin/openid.3.5.0.zip" -o "openid.zip"
+      unzip "openid.zip"
+      rm "openid.zip"
+      # Latest YoastSEO does not support 5.9.3 version of WordPress.
+      curl -sSL "https://downloads.wordpress.org/plugin/wordpress-seo.18.9.zip" -o "wordpress-seo.zip"
+      unzip "wordpress-seo.zip"
+      rm "wordpress-seo.zip"
+      git clone https://git.launchpad.net/~canonical-sysadmins/wordpress-launchpad-integration/+git/wordpress-launchpad-integration wordpress-launchpad-integration
+      git clone https://git.launchpad.net/~canonical-sysadmins/wordpress/+git/openstack-objectstorage-k8s openstack-objectstorage-k8s
+      git clone https://git.launchpad.net/~canonical-sysadmins/wordpress-teams-integration/+git/wordpress-teams-integration wordpress-teams-integration
+      git clone https://git.launchpad.net/~canonical-sysadmins/wordpress/+git/wp-plugin-xubuntu-team-members xubuntu-team-members
+      rm -rf */.git)
+
+      # Install wordpress themes
+      (cd wordpress_install_dir/wp-content/themes;
+      git clone https://git.launchpad.net/~canonical-sysadmins/ubuntu-community-webthemes/+git/light-wordpress-theme light-wordpress-theme
```

Comment:
I know that this works but it's a bit circling around the interesting thing that rockcraft has (vs dockerfiles): parts.  
If you defined all those clones as separate parts, you would gain a lot of time when rebuilding because they're will be cached (and eventually could be done in parallel).

You can look at the discourse's one as an example for this: https://github.com/canonical/discourse-k8s-operator/blob/main/discourse_rock/rockcraft.yaml#L94

Ideally, we want to have a part for each individual unit of things that we do (would it be importing files, changing them, etc..)

Reply:
Thanks for the tip, Initially I was hesitating putting each plugin/theme into a separate part just because of the sheer number of parts and dependencies between parts that I'd need to manage. I have updated the rockfile to do that and it seems pretty manageable so all is good :smile: